src/platform/site-wide/user-nav/containers/Main.jsx

```diff
@@ -38,18 +42,10 @@ export class Main extends React.Component {
 
   componentDidUpdate() {
     const { currentlyLoggedIn, showLoginModal } = this.props;
-    const nextParam = this.getRedirectUrl();
-
-    const shouldRedirect =
-      currentlyLoggedIn && nextParam && !window.location.pathname.includes('verify');
-
-    if (shouldRedirect) {
-      const redirectPath = nextParam.startsWith('/') ? nextParam : `/${nextParam}`;
-      window.location.replace(redirectPath);
-    }
-
     const shouldCloseLoginModal = currentlyLoggedIn && showLoginModal;
 
+    if (currentlyLoggedIn) this.executeRedirect();
+
     if (shouldCloseLoginModal) { this.props.toggleLoginModal(false); }
```

Comment:
The removed code was largely moved into `executeRedirect`, which redirects the user if the URL warrants it.